src/components/_classes/component/Component.js

```diff
@@ -2961,12 +2961,20 @@ export default class Component extends Element {
     }
     for (const i in this.refs.input) {
       if (this.refs.input.hasOwnProperty(i)) {
-        this.setValueAt(i, isArray ? value[i] : value, flags);
+        this.setValueAt(i, isArray && !this.isSingleInputValue() ? value[i] : value, flags);
```

Comment:
Isn't the core problem here that I didn't account for JSON text area types in the `normalizeValue` method of the `Multivalue` class? I'd rather just fix that function than spread this type of normalization logic around

Reply:
This change is not related to the value normalization. When the value is being set, line 2962 iterates though the input references and calls setValueAt for each input. When the value is an array, it takes the " i" element of the array and sets it to the "i" input. But this does not work for textArea with array value. For it, we should set full array into the first input otherwise only the first array element is displayed.